File: Exercise/Hard/13. Excel Column Number/Code/Excel Column Number.java
import java.util.Scanner;

public class ExcelColumnNumber {

    public static int titleToNumber(String s) {
        int result = 0;
        for (char c : s.toCharArray()) {
            result = result * 26 + (c - 'A' + 1);
        }
        return result;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.nextLine();
        
        int number = titleToNumber(s);
        System.out.println(number);
        
        scanner.close();
    }
}

File: Exercise/Hard/13. Excel Column Number/Code/Excel Column Number.c
#include <stdio.h>
#include <string.h>

int titleToNumber(char *s) {
    int result = 0;
    while (*s) {
        result = result * 26 + (*s - 'A' + 1);
        s++;
    }
    return result;
}

int main() {
    char s[100];
    scanf("%s", s);
    
    int number = titleToNumber(s);
    printf("%d\n", number);
    
    return 0;
}

